### PR TITLE
NAS-108093 / 12.0 / Create VM context if libvirt fails to poweroff/shutdown vm in bhyve

### PR DESCRIPTION
**Pull request by sonicaj**

### CODE REVIEW

File: src/middlewared/middlewared/plugins/vm.py [hunk 1]
```diff
@@ -270,7 +276,23 @@ def restart(self, vm_data=None, shutdown_timeout=None):
...
+        except libvirt.libvirtError:
+            if not self.destroy_vm_context():
+                raise
+
+    def destroy_vm_context(self):
+        # There is a race condition where poweroff/shutdown fail sometimes in libvirt which results in it trying
+        # to destroy a VM whose context does not exist resulting in user having to reboot their machines in order
+        # to get the VM back.
+        # This is a workaround where we create a vm context if it doesn't exist so stop/poweroff work
+        if os.path.exists('/dev/vmm') and self.libvirt_domain_name in os.listdir('/dev/vmm'):
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/vm.py:290]: Did you mean libvirt_domain_name `not in`?

Comment (william-gr) [reply src/middlewared/middlewared/plugins/vm.py:290]: Guess not, nvm

Comment (sonicaj) [reply src/middlewared/middlewared/plugins/vm.py:290]: Nope, if the context is there - then the error is different which we have been seeing reported by users

Comment (sonicaj) [reply src/middlewared/middlewared/plugins/vm.py:290]: In that case we should raise whatever the error is

**Review by themylogin — APPROVED** [review]

**Pull request merged**